ingestion/src/metadata/ingestion/source/database/bigquery/connection.py

```diff
@@ -170,3 +171,28 @@ def test_connection_inner(engine):
         )
 
     test_connection_inner(engine)
+
+
+def get_table_view_names(connection, schema=None):
+    with connection.connect() as conn:
+        current_schema = schema
+        client = conn.connection._client
+        item_types = ["TABLE", "EXTERNAL", "VIEW", "MATERIALIZED_VIEW"]
+        datasets = client.list_datasets()
+        result = []
+        for dataset in datasets:
+            if current_schema is not None and current_schema != dataset.dataset_id:
+                continue
+
+            try:
+                tables = client.list_tables(dataset.reference, page_size=1)
+                for table in tables:
+                    if table.table_type in item_types:
+                        break
+            except NotFound:
+                # It's possible that the dataset was deleted between when we
+                # fetched the list of datasets and when we try to list the
+                # tables from it. See:
+                # https://github.com/googleapis/python-bigquery-sqlalchemy/issues/105
+                pass
```

Comment:
I suggest adding a debug message here so that if we hit this edge case we get some signal.

Reply:
I think it should be fine, because this is a replica of function within bigquery dialect

in the test connection we can have the logger but the actual function would just pass it anyways 🤔